### PR TITLE
Add docs for analyzers ASP0011-14

### LINKED ISSUE #27020: Add analyzers introduced in .NET 7
See https://github.com/dotnet/aspnetcore/issues/44009

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 657f3da7-ec96-7d56-f1b4-4787e0f0fd84
* Version Independent ID: 657f3da7-ec96-7d56-f1b4-4787e0f0fd84
* Content: [Code analysis in ASP.NET Core apps](https://docs.microsoft.com/en-us/aspnet/core/diagnostics/code-analysis?view=aspnetcore-6.0)
* Content Source: [aspnetcore/diagnostics/code-analysis.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/diagnostics/code-analysis.md)
* Product: **aspnet-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by captainsafia**
Closes #27020

### CODE REVIEW

**Comment by guardrex** [comment]
btw @captainsafia ... This one and the last few are throwing ...

> Line 6: [Suggestion] Invalid value for 'ms.author', 'captainsafia' is not a valid Microsoft alias.

They're listed on the build report.

**Comment by captainsafia** [comment]
> btw @captainsafia ... This one and the last few are throwing ...
> 
> > Line 6: [Suggestion] Invalid value for 'ms.author', 'captainsafia' is not a valid Microsoft alias.
> 
> They're listed on the build report.

Oops! I see this now. It wasn't failing the build check so I assumed it was alright. What's the correct way to go about fixing this? Use my MS alias (safia)?

**Comment by guardrex** [comment]
Yes, I think that's going to take care of it.

File: aspnetcore/diagnostics/asp0011.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+| | Value |
+|-|-|
+| **Rule ID** |ASP0011|
+| **Category** |Usage|
+| **Fix is breaking or non-breaking** |Non-breaking|
+
+## Cause
+
+`ConfigureLogging` is not the recommended strategy for configuring logging in a minimal API application.
```
Comment (Rick-Anderson) [inline aspnetcore/diagnostics/asp0011.md:20]:
```suggestion
`ConfigureLogging` isn't the recommended strategy for configuring logging in a minimal API application.
```

File: aspnetcore/diagnostics/asp0011.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+## Rule description
+
+`ConfigureLogging` is not the recommended strategy for configuring logging in a minimal API application.
```
Comment (Rick-Anderson) [inline aspnetcore/diagnostics/asp0011.md:24]:
```suggestion
`ConfigureLogging` isn't the recommended strategy for configuring logging in a minimal API application.
```

File: aspnetcore/diagnostics/asp0011.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+})
+
+var app = builder.Build();
+
+app.Run();
+```
+
+## How to fix violations
+
+To fix a violation of this rule, leverage the `Logging` property on the `WebApplicationBuilder` to modify the logging configuration directly without the need for an additional `ConfigureLogging` call.
```
Comment (Rick-Anderson) [inline aspnetcore/diagnostics/asp0011.md:41]:
```suggestion
To fix a violation of this rule, use the `Logging` property on the `WebApplicationBuilder` to modify the logging configuration directly without the need for an additional `ConfigureLogging` call.
```

File: aspnetcore/diagnostics/asp0012.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+| | Value |
+|-|-|
+| **Rule ID** |ASP0012|
+| **Category** |Usage|
+| **Fix is breaking or non-breaking** |Non-breaking|
+
+## Cause
+
+`ConfigureServices` is not the recommended strategy for registering services in DI in a minimal API application.
```
Comment (Rick-Anderson) [inline aspnetcore/diagnostics/asp0012.md:20]:
```suggestion
`ConfigureServices` isn't the recommended strategy for registering services in DI in a minimal API application.
```

File: aspnetcore/diagnostics/asp0012.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+## Rule description
+
+`ConfigureServices` is not the recommended strategy for configuring logging in a minimal API application.
```
Comment (Rick-Anderson) [inline aspnetcore/diagnostics/asp0012.md:24]:
```suggestion
`ConfigureServices` isn't the recommended strategy for configuring logging in a minimal API application.
```

File: aspnetcore/diagnostics/asp0012.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+})
+
+var app = builder.Build();
+
+app.Run();
+```
+
+## How to fix violations
+
+To fix a violation of this rule, leverage the `Services` property on the `WebApplicationBuilder` to modify the DI container directly without the need for an additional `ConfigureServices` call.
```
Comment (Rick-Anderson) [inline aspnetcore/diagnostics/asp0012.md:41]:
```suggestion
To fix a violation of this rule, use the `Services` property on the `WebApplicationBuilder` to modify the DI container directly without the need for an additional `ConfigureServices` call.
```

File: aspnetcore/diagnostics/asp0013.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+| | Value |
+|-|-|
+| **Rule ID** |ASP0013|
+| **Category** |Usage|
+| **Fix is breaking or non-breaking** |Non-breaking|
+
+## Cause
+
+`Configure` is not the recommended strategy for reading and writing to configuration in a minimal API application.
```
Comment (Rick-Anderson) [inline aspnetcore/diagnostics/asp0013.md:20]:
```suggestion
`Configure` isn't the recommended strategy for reading and writing to configuration in a minimal API application.
```

File: aspnetcore/diagnostics/asp0013.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+## Rule description
+
+`Configure` is not the recommended strategy for configuring logging in a minimal API application.
```
Comment (Rick-Anderson) [inline aspnetcore/diagnostics/asp0013.md:24]:
```suggestion
`Configure` isn't the recommended strategy for configuring logging in a minimal API application.
```

File: aspnetcore/diagnostics/asp0013.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+})
+
+var app = builder.Build();
+
+app.Run();
+```
+
+## How to fix violations
+
+To fix a violation of this rule, leverage the `Configuration` property on the `WebApplicationBuilder` to modify application configuration directly without the need for an additional `ConfigureAppConfiguration` call.
```
Comment (Rick-Anderson) [inline aspnetcore/diagnostics/asp0013.md:41]:
```suggestion
To fix a violation of this rule, use the `Configuration` property on the `WebApplicationBuilder` to modify application configuration directly without the need for an additional `ConfigureAppConfiguration` call.
```

File: aspnetcore/diagnostics/asp0014.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+| **Category** |Usage|
+| **Fix is breaking or non-breaking** |Non-breaking|
+
+## Cause
+
+Routes can be registered directly at the top-level of a minimal API application.
+
+## Rule description
+
+Routes can be registered directly at the top-level of a minimal API application and do not need to be nested within a `UseEndpoints` call.
```
Comment (Rick-Anderson) [inline aspnetcore/diagnostics/asp0014.md:23]:
```suggestion
Routes can be registered directly at the top-level of a minimal API application and don't need to be nested within a `UseEndpoints` call.
```

**Review by Rick-Anderson — APPROVED** [review]

**Pull request merged**